ConcordiumWallet/Views/Stake/Baking/ComissionSettings/BakerCommissionSettingsView.swift

```
@@ -0,0 +1,116 @@
+                Text(error.errorMessage)
+            })
+        .onAppear { viewModel.fetchData() }
+        .padding()
+    }
+}
+
+struct BakerCommissionSliderView: View {
+    var range: CommissionRange
+    @Binding var commission: Double
+    let formatter: NumberFormatter = .comissionFormatter
+
+    var body: some View {
+        let commissionBinding = Binding<Double>(get: {
+            self.commission
+        }, set: {
+            // Rounding for edge cases of sliders to prevent odd values caused by floating point errors.
+            if $0 + BakerCommissionSettingsView.sliderStep >= range.max {
+                commission = range.max
+            } else if $0 - BakerCommissionSettingsView.sliderStep <= range.min {
+                commission = range.min
+            } else {
+                self.commission = $0
+            }
+        })
+        HStack {
+            VStack {
+                Text("Min:")
+                Text("\(formatter.string(from: NSNumber(value: range.min)) ?? " - ")%")
+            }
+            VStack {
+                HStack(alignment: .center, spacing: 1) {
+                    Spacer()
+                    TextField("", value: $commission, formatter: NumberFormatter.comissionFormatter)
+                        .disabled(range.min == range.max)
+                        .keyboardType(.decimalPad)
+                        .textFieldStyle(.roundedBorder)
+                        .lineLimit(1)
+                        .multilineTextAlignment(.center)
+                        .frame(width: 72)
+                    Text(" %")
+                    Spacer()
+                }
+                if range.min < range.max {
+                    Slider(value: commissionBinding, in: range.min ... range.max, step: BakerCommissionSettingsView.sliderStep)
+                }
+            }
+            VStack {
+                Text("Max:")
+                Text("\(formatter.string(from: NSNumber(value: range.max)) ?? " - ")%")
```

How could `formatter.string` return `nil`? If never, maybe we should just force unwrap? Displaying ` - %` doesn't seem very informative.

---

it might if nsnumber is not a number. for instance when range.max value is not a number as string

---

If you mean like `range.max = NaN` then I think force unwrap is in order as nothing is going to work in that case. Failing silently is the worst we can do IMHO.